From df6262dc76a74baaec250a53a5a99fc5a663df8a Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@debian.org>
Date: Thu, 3 Apr 2014 11:01:26 +0100
Subject: Fix device_id scanning on 32-bit systems

scsi_dev_data->device_id is a u64, so should be scanned with %llX rather
than %lX.  On 32-bit big-endian systems I believe this will scan the
device_id into the wrong half of the 64-bit value.

Forwarded: yes
Last-Update: 2014-04-03

Patch-Name: fix-device_id-scanning.patch
---
 iprlib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iprlib.c b/iprlib.c
index 3e4ac2d..ea5397e 100644
--- a/iprlib.c
+++ b/iprlib.c
@@ -5471,7 +5471,7 @@ int get_scsi_dev_data(struct scsi_dev_data **scsi_dev_ref)
  
 		len = sysfs_read_attr(devpath, "device_id", buff, 256);
 		if (len > 0)
-			sscanf(buff, "%lX", &scsi_dev_data->device_id);
+			sscanf(buff, "%llX", &scsi_dev_data->device_id);
 
 		strcpy(scsi_dev_data->dev_name,"");
 		strcpy(scsi_dev_data->gen_name,"");
